lib/s3/storage.go

```diff
@@ -67,7 +87,15 @@ func (s *Storage) List(path string) ([]string, error) {
 		}
 	}
 
-	return result, nil
+	return result
+}
+
+func prefixes(result []string, res *s3.ListObjectsOutput) []string {
```

Comment:
I think we can make this function more `pure` by doing something like:
```go
func prefixes(res *s3.ListObjectsOutput) []string {
	result := make([]string, 0)

	for _, prefix := range res.CommonPrefixes {
		result = append(result, path.Base(*prefix.Prefix))
	}

	return result
}
```
Then we can do something like this in the `List` function:
```go
result = append(result, prefixes(res)...)
```

Reply:
Perfect idea! Done.